Automattic/wp-calypso
# Domains: Replace "Change Site Address" page with modal

#### Changes proposed in this Pull Request
This PR removes the old-style "change site address" page and replaces it with a modal in the domain management page. For all other pages, the user is not redirected to the domain management page. It also fixes a small bug where .blog subdomains were shown incorrectly in the address change confirmation modal. There were no changes made to the feature itself.

#### Preview
##### Before
https://user-images.githubusercontent.com/18705930/152861536-0847805e-6f53-4e1e-a406-bc70c0f6d7a5.mov

##### After
https://user-images.githubusercontent.com/18705930/152861553-6c33083c-2c1d-4695-b830-fb55313b9adb.mov

#### Testing instructions
- Browse to `/domains/manage`;
- Select the "Change site address" option in the free domain item component's 3 dot menu (the `Your free WordPress.com address is (...)` element on the footer);
- Check that a modal is displayed instead of the previous page;
- Check that the change site feature still works the same as previously:
  - Confirm that you get an error if you try to use an unavailable address;
  - Confirm that you are prompted for confirmation before changing the site address;
    - Check that both cancel and confirm (change site address) options are still working as intended.
